test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -110,252 +94,292 @@ public void addUnit() {
+        is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+    assertThat(unitCollection.getUnitCount(unitTypeOne, otherPlayerID), is(equalTo(unitCountOtherPlayerUnitTypeOne)));
+    assertThat(unitCollection.getUnitCount(unitTypeTwo, otherPlayerID), is(equalTo(unitCountOtherPlayerUnitTypeTwo)));
   }
 
   @Test
   public void getUnitCountByPlayerID() {
-    assertThat(uc.getUnitCount(defaultPlayerID), is(equalTo(0)));
-    assertThat(uc.getUnitCount(otherPlayerID), is(equalTo(0)));
+    assertThat(unitCollection.getUnitCount(defaultPlayerID), is(equalTo(0)));
+    assertThat(unitCollection.getUnitCount(otherPlayerID), is(equalTo(0)));
 
-    addAllDefaultPlayerUnits();
-    assertThat(uc.getUnitCount(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
-    assertThat(uc.getUnitCount(otherPlayerID), is(equalTo(0)));
+    addAllDefaultPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.getUnitCount(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
+    assertThat(unitCollection.getUnitCount(otherPlayerID), is(equalTo(0)));
 
-    addAllOtherPlayerUnits();
-    assertThat(uc.getUnitCount(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
-    assertThat(uc.getUnitCount(otherPlayerID), is(equalTo(unitCountOtherPlayer)));
+    addAllOtherPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.getUnitCount(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
+    assertThat(unitCollection.getUnitCount(otherPlayerID), is(equalTo(unitCountOtherPlayer)));
   }
 
   @Test
   public void containsAll() {
     final Collection<Unit> unitsOfDefaultPlayerOfUnitTypeOne = getUnitsOfDefaultPlayerOfUnitTypeOne();
-    assertThat(uc.containsAll(unitsOfDefaultPlayerOfUnitTypeOne), is(equalTo(false)));
-    addAllDefaultPlayerUnits();
-    assertThat(uc.containsAll(unitsOfDefaultPlayerOfUnitTypeOne), is(equalTo(true)));
+    assertThat(unitCollection.containsAll(unitsOfDefaultPlayerOfUnitTypeOne), is(equalTo(false)));
+    addAllDefaultPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.containsAll(unitsOfDefaultPlayerOfUnitTypeOne), is(equalTo(true)));
   }
 
   @Test
   public void getUnitsByUnitTypeAndMaxValue() {
-    addAllDefaultPlayerUnits();
-    assertThat(uc.getUnits(unitTypeTwo, Integer.MAX_VALUE).size(), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
-    assertThat(uc.getUnits(unitTypeTwo, unitCountDefaultPlayerUnitTypeTwo).size(),
+    addAllDefaultPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.getUnits(unitTypeTwo, Integer.MAX_VALUE).size(),
+        is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+    assertThat(unitCollection.getUnits(unitTypeTwo, unitCountDefaultPlayerUnitTypeTwo).size(),
         is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
   }
 
   @Test
   public void getUnitsByType() {
-    addAllDefaultPlayerUnits();
-    final IntegerMap<UnitType> unitsByType = uc.getUnitsByType();
+    addAllDefaultPlayerUnitsToUnitCollection();

```

Comment:
If a method needs to alter setup data, and we do not do that setup in @Setup, then doing the entire data setup in the @Test method is good. When reading this, I'm saying to myself, "please don't make me go and find this method so I can figure out what it does". Hunting around  a lot makes it hard to go through code. If on the other hand you see that the data is being formed, and you see the process from start to end, then you know roughly what any helper method would be doing, since you can see the outputs.